build/citations.py

```diff
@@ -50,9 +50,10 @@ def semicolon_separate_references(text):
 def get_brackets_without_reference(text):
     """
     Find bracketed text that does not start with @. Does not match
-    brackets that are followed by an open parenthesis.
+    brackets that are followed by an open parenthesis. Also ignore
+    brackets that start with ! for captions.
     """
-    pattern = re.compile(r'(\[[^@][^\]]*?\])[^(]', flags=re.DOTALL)
+    pattern = re.compile(r'(?<!\!)(\[(?!\@).+\])[^(]', flags=re.DOTALL)
```

Comment:
This part will hopefully be superseded if we address #2 and no longer have to worry about brackets at all?